dll/win32/shell32/wine/control.c

```diff
@@ -821,26 +821,29 @@ Control_ShowAppletInTaskbar(CPlApplet* applet, UINT index)
 {
     ITaskbarList* pTaskbar = NULL;
 
-    SetWindowTextW(applet->hWnd, applet->info[index].name);
-    SendMessageW(applet->hWnd, WM_SETICON, ICON_SMALL, (LPARAM)applet->info[index].icon);
-
-    /* Add button to the taskbar */
-    ShowWindow(applet->hWnd, SW_SHOWMINNOACTIVE);
-
-    /* Activate the corresponding button in the taskbar */
-    CoInitialize(NULL);
-    if (CoCreateInstance(&CLSID_TaskbarList,
-                         NULL, CLSCTX_INPROC_SERVER,
-                         &IID_ITaskbarList,
-                         (LPVOID*)&pTaskbar) == S_OK)
+    if (GetParent(applet->hWnd) == NULL) // Is desktop the parent window?
     {
```

Comment:
By the way, why is it necessary to call the `SetWindowTextW` function? In principle the applet's window will set its title, so it should not be force-changed manually like this.
Besides, the set-icon call below can be ill-defined. For example, the "icon" handle can be NULL:
![image](https://user-images.githubusercontent.com/1969829/166987273-a1572604-dbc6-4f26-bcdb-3cd2df5d7060.png)

Reply:
Without these function calls it looks like this.
![nocall](https://user-images.githubusercontent.com/102076097/167025771-4f2706d9-770c-4f74-8874-26a01927808e.png)

Reply:
In case of an empty icon handle, default icon for taskbar is used automatically.